src/pages/contests/index.js

```diff
@@ -1,57 +1,197 @@
-import React, { useState } from "react";
+import React, { useEffect, useState } from "react";
 import { graphql } from "gatsby";
 
-import { contestsByState } from "../../utils/filter";
-
 import ContestList from "../../components/ContestList";
 import DefaultLayout from "../../templates/DefaultLayout";
 
 export default function Contests({ data }) {
-  // @todo: implement global state management instead of props drilling
+  const [filteredContests, setFilteredContest] = useState(null);
   const [contestStatusChanges, updateContestStatusChanges] = useState(0);
-
+  const [isLoading, setIsLoading] = useState(false);
+  const contests = data.contests.edges;
   const updateContestStatus = () => {
     updateContestStatusChanges(contestStatusChanges + 1);
   };
 
-  const contests = data.contests.edges;
+  const sortContests = (array, status) => {
+    let statusObject = {
+      upcomingContests: [],
+      activeContests: [],
+      sponsorReview: [],
+      judging: [],
+      awarding: [],
+      reporting: [],
+      completed: [],
+      other: [],
+    };
+
+    array.forEach((element) => {
+      const statusAndIdObj = status.filter(
+        (el) => el.contestId === element.node.contestid
+      );
+      if (statusAndIdObj === []) {
+        return null;
+      }
+      const data = {
+        ...element.node,
+        status: statusObject[0]?.status,
+      };
+      switch (statusAndIdObj[0]?.status) {
+        case "Pre-Contest":
+          statusObject.upcomingContests.push(data);
+          break;
+        case "Preview week":
+          statusObject.upcomingContests.push(data);
+          break;
+        case "Active Contest":
+          statusObject.activeContests.push(data);
+          break;
+        case "Active":
+          statusObject.activeContests.push(data);
+          break;
+        case "Sponsor Review":
+          statusObject.sponsorReview.push(data);
+          break;
+        case "Needs Judging":
+          statusObject.judging.push(data);
+          break;
+        case "Judging Complete":
+          statusObject.judging.push(data);
+          break;
+        case "Awarding":
+          statusObject.awarding.push(data);
+          break;
+        case "Reporting":
+          statusObject.reporting.push(data);
+          break;
+        case "Completed":
+          statusObject.completed.push(data);
+          break;
+        default:
+          statusObject.other.push(data);
+          break;
+      }
+    });
+    for (const keys in statusObject) {
+      statusObject[keys].sort(function (a, b) {
+        let keyA = new Date(a.start_time);
+        let keyB = new Date(b.start_time);
+        if (keyA < keyB) return -1;
+        if (keyA > keyB) return 1;
+        return 0;
+      });
+    }
+    return statusObject;
+  };
 
-  const filteredContests = contestsByState({ contests });
+  useEffect(() => {
+    setIsLoading(true);
+    fetch("/.netlify/functions/getNotionData")
```

Comment:
Something like the following fits more with the current naming convention (to the extent that's a thing):

```suggestion
    fetch("/.netlify/functions/get-contest")
```

Reply:
Totally ! I'll change it